x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/table/add_data_popover.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiBadge,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPopover,
+  EuiPopoverFooter,
+  EuiPopoverTitle,
+  EuiText,
+} from '@elastic/eui';
+import {
+  ObservabilityOnboardingLocatorParams,
+  OBSERVABILITY_ONBOARDING_LOCATOR,
+} from '@kbn/deeplinks-observability';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { useKibanaContextForPlugin } from '../../../../../hooks/use_kibana';
+
+export const AddDataPopover = () => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const {
+    services: { share },
+  } = useKibanaContextForPlugin();
+  const addDataLinkHref = share.url.locators
+    .get<ObservabilityOnboardingLocatorParams>(OBSERVABILITY_ONBOARDING_LOCATOR)
+    ?.getRedirectUrl({ category: 'logs' });
+
+  const troubleshootingLinkHref = 'https://ela.st/host-troubleshooting';
+
+  const popoverContent = {
+    title: i18n.translate('xpack.infra.addDataPopover.wantToSeeMorePopoverTitleLabel', {
+      defaultMessage: 'Want to see more?',
+    }),
+    content: i18n.translate('xpack.infra.addDataPopover.understandHostPerformanceByTextLabel', {
+      defaultMessage: 'Understand host performance by collecting more metrics.',
+    }),
+    button: i18n.translate('xpack.infra.addDataPopover.understandHostPerformanceByTextLabel', {
+      defaultMessage: 'Add data',
+    }),
+    link: i18n.translate('xpack.infra.addDataPopover.troubleshootingLinkLabel', {
+      defaultMessage: 'Troubleshooting',
+    }),
+  };
+
+  const onButtonClick = () => setIsPopoverOpen((prevIsPopoverOpen) => !prevIsPopoverOpen);
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const button = (
```

Comment:
Nit: Do you reuse the button somewhere? I think we can pass it directly in the `return` if we don't use it somewhere else